src/components/Contacts/ContactDetails/ContactDontationsTab/ContactDonationsList/ContactDonationsList.tsx

```diff
@@ -0,0 +1,138 @@
+    data,
+    loading,
+    fetchMore,
+    networkStatus,
+  } = useContactDonationsListQuery({
+    variables: {
+      accountListId: accountListId,
+      contactId: contactId,
+      first: range,
+    },
+    notifyOnNetworkStatusChange: true,
+  });
+
+  const { t } = useTranslation();
+
+  return (
+    <Box>
+      {loading || networkStatus === 3 ? (
+        <>
+          <DonationLoadingPlaceHolder />
+          <DonationLoadingPlaceHolder />
+          <DonationLoadingPlaceHolder />
+        </>
+      ) : (
+        <>
+          <Table>
+            <TableHead>
+              <TableCell>{t('Date')}</TableCell>
+              <TableCell>{t('Amount')}</TableCell>
+              <TableCell>{t('Converted Amount')}</TableCell>
+            </TableHead>
+            {data?.contact !== null &&
+            data?.contact.donations.nodes !== undefined ? (
+              data?.contact.donations.nodes.map((donation) => (
+                <TableRow key={donation.id}>
+                  <TableCell>
+                    {DateTime.fromISO(donation.donationDate).toLocaleString()}
+                  </TableCell>
+                  <TableCell>
+                    {currencyFormat(
+                      donation.amount.amount,
+                      donation.amount.currency,
+                    )}
+                  </TableCell>
+                  <TableCell>
+                    {currencyFormat(
+                      donation.amount.convertedAmount,
+                      donation.amount.convertedCurrency,
+                    )}
+                  </TableCell>
+                </TableRow>
+              ))
+            ) : (
+              <></>
+            )}
+          </Table>
+          {!loading &&
+          data?.contact.donations.pageInfo !== undefined &&
+          data?.contact.donations.pageInfo.hasNextPage === true ? (
+            <LoadMoreButton
+              variant="outlined"
+              onClick={() => {
+                fetchMore({
+                  variables: {
+                    first: range,
+                    after: data.contact.donations.pageInfo.endCursor,
+                  },
+                  updateQuery: (prev, { fetchMoreResult }) => {
+                    if (!fetchMoreResult) {
+                      return prev;
+                    }
+                    return {
+                      contact: {
+                        donations: {
+                          nodes: [
+                            ...prev.contact.donations.nodes,
+                            ...fetchMoreResult.contact.donations.nodes,
+                          ],
+                          pageInfo: fetchMoreResult.contact.donations.pageInfo,
+                          totalCount:
+                            fetchMoreResult.contact.donations.totalCount,
+                          totalPageCount:
+                            fetchMoreResult.contact.donations.totalPageCount,
+                        },
+                      },
```

Comment:
You are probably only loading one root key in the query here but spreading every level is safer if more was added to the query in the future. Also by spreading `fetchMoreResult.contact.donations`, you don't need to manually set all the pagination metadata.
```suggestion
                      ...prev
                      ...fetchMoreResult,
                      contact: {
                        ...prev.contact,
                        ...fetchMoreResult.contact,
                        donations: {
                          ...prev.contact.donations,
                          ...fetchMoreResult.contact.donations,
                          nodes: [
                            ...prev.contact.donations.nodes,
                            ...fetchMoreResult.contact.donations.nodes,
                          ],
                        },
                      },
```